libflux/src/core/semantic/types.rs

```diff
@@ -432,21 +434,26 @@ impl MonoType {
                 }),
             },
             MonoType::Duration => match with {
-                Kind::Comparable | Kind::Equatable | Kind::Nullable | Kind::Negatable => {
-                    Ok(Substitution::empty())
-                }
+                Kind::Comparable
```

Comment:
@jlapacik would be a better person to look at this, but it looks correct to me.

Reply:
This is correct.